test/qa-tests/jstests/export/type_case.js

```diff
@@ -33,7 +33,7 @@
     '--type="foobar"',
     '--fields', 'a']
     .concat(commonToolArgs));
-  assert.eq(3, ret);
+  assert.eq(1, ret);
```

Comment:
This was changed because MongoExport.validateSettings is now taken care of by MongoExport.New, so failures from that method return with util.ExitError instead of util.ExitBadOptions.

Reply:
it seems like we might still want the original error code if it's possible, since it better conveys the problem.

Reply:
I agree the original error code was better. The problem was that validateSettings should be called by New, but if New returns an err, it's not possible to know where that error came from. I've added a SetupError type in the new commits to convey what the error code should be when exiting.